x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all/index.tsx

```diff
@@ -111,6 +113,10 @@ export const AllRules = React.memo<AllRulesProps>(
     const { loading: isLoadingRulesStatuses, rulesStatuses } = useRulesStatuses(rules);
     const history = useHistory();
     const [, dispatchToaster] = useStateToaster();
+    const mlCapabilities = useMlCapabilities();
+
+    const hasMlPermissions =
+      mlCapabilities.isPlatinumOrTrialLicense && hasMlAdminPermissions(mlCapabilities);
```

Comment:
I've waffled on this, but since we're going to be better defining what ML Roles/Permissions mean in 7.8 I'm going to leave as is for this PR in effort to keep things explicit (+FF & all).

I have updated the source with a TODO though, so thank you for calling attention to this code smell... 💩